[18.0][MIG] document_page_approval: Migration to 18.0

BhaveshHeliconia opened the pull request.

cloutierjo committed:
create document_page_approval module
(../7.0pageapproval/ rev 1)

fix status bar in page approval
(../7.0pageapproval/ rev 2)

change approval workflow in page history and started to add showing approved page in normal page
(../7.0pageapproval/ rev 3)

show last approved content and aproved by
(../7.0pageapproval/ rev 4)

Fix bug on page creation
(../7.0pageapproval/ rev 5)

add status in history tree view
(../7.0pageapproval/ rev 6)

add a need approval field and hide the workflow when no approval needed
(../7.0pageapproval/ rev 7)

add access validation to approval
(../7.0pageapproval/ rev 8)

give rigth to Page aprover group on the page_history model and hide the button for unallowed group thought the security isn't enforced a the orm level
(../7.0pageapproval/ rev 9)

made invisible the prover group when not needed
(../7.0pageapproval/ rev 10)

add email to aprovers group's members
(../7.0pageapproval/ rev 11)

Commit:
[IMP] Documentation
[ADD] Translation file + french
[FIX] employee_id field does not exist
[FIX] user_email field deprecated
[IMP] Split long lines
[ADD] Images

Commit:
Set document_page_approval as unported

gborelli committed:
start porting

fix act_draft and minor ui fix

remove unused variables

change order of xml data files

llacroix committed:
Comment workflow that cause the error

It should work but further errors could help understand what's going

Commented other necessary part

Add back workflow

Add dependency to knowledge

Commit:
[IMP] - Changed string from email template.
[IMP] - Improved views to match document_page improvements.
[IMP] - Added notification to followers when a new version is approved.
[ADD] - Updated document_page_approval pot and added es translations.
[IMP] - document_page_approval: Updated spanish translation.

Commit:
[FIX] - document-page-approval: Fixes #60 url not valid when website module is installed.

naousse committed:
[MIG] document_page_approval: Migration to 9.0

* Lint error corrected
* test improved
* lint error corrected in test file
* readme corrected
* test improved
* test of document page history workflow added
* CI error corrected
* dreipst comment
* Migration to new api improved in model's file and data tag remove in xml file
* code totally migrate to odoo 9.0 api
* test bug fixed
* test error fixed
* Lint error improved

LeartS committed:
[MIG] document_page_approval: Migration to 10.0

ivantodorovich committed:
[IMP][10.0][document_page_approval] Change Requests and workflow improvements on documents (#155)

max3903 committed:
[FIX] View

ivantodorovich committed:
[FIX+IMP] document_page_approval:
* Rename classes and files to match model names, and removed workflow engine.
* Some fields should be readonly if not in draft.
* Allow Knowledge / Manager to edit previous history names and summary
* Improve ux by adding user_has_drafts to document pages.
* Changes because of implementation of active field on document_page
* Add ir.rules for document.page.history
* Allow to quickly cancel&draft from 'to approve' state
* FIX BUG: am_i_approver was being run as sudo(), hence always giving the user Approving rights.
* FIX hooks
  Because of default='draft' on the state field, state IS NULL wasn't selecting anything.
  Left the condition in case this is changed in the future.

Commit:
[MIG] document_page_approval: Migration to 11.0

MiquelRForgeFlow committed:
[FIX] pylint/javascript-lint

[ADD] icon.png

nicomacr committed:
[FIX] document_page_aproval: Modify in records rules

 Add the the group eployee to the rule "Change Request Global" to allow to the user with the rule "Change Request Approver" to create pages

Commit:
[MIG] document_page_approval: Migration to v12

[UPD] Update document_page_approval.pot

Commit:
[IMP] document_page_approval: black, isort

Commit:
Migrate to 13.0 document_page_approval

To pass test is necesary set approval required to False in demo data

jeroen-vet committed:
[IMP] document_page_approval: black, isort, prettier + squashed admin commits

victoralmau committed:
[FIX] document_page_approval: Add correct followers in history pages

[BOT] post-merge updates

jeroen-vet committed:
[MIG] document_page_approval: Migration to 14.0

[UPD] Update document_page_approval.pot

[UPD] README.rst

pedrobaeza committed:
[FIX] document_page_approval: Avoid W7940 (dangerous-view-replace-wo-priority)

victoralmau committed:
[MIG] document_page_approval: Migration to 15.0
TT36950

[UPD] Update document_page_approval.pot

[UPD] README.rst

FernandoRomera committed:
[16.0][MIG] document_page_approval: Migration to 16.0

[UPD] Update document_page_approval.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 83.6% (51 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/hr/

Translated using Weblate (Croatian)

Currently translated at 96.7% (59 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/hr/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

victoralmau committed:
[FIX] document_page_approval: Change mail template expressions

[UPD] Update document_page_approval.pot

document_page_approval 16.0.1.1.0

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 96.7% (59 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

Translated using Weblate (Italian)

Currently translated at 96.7% (59 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

[UPD] Update document_page_approval.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 96.7% (59 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

luis-ron committed:
Translated using Weblate (Spanish)

Currently translated at 95.0% (58 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/es/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 96.7% (59 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/es/

[UPD] Update document_page_approval.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (61 of 61 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/es/

[UPD] Update document_page_approval.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

peterromao committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (60 of 60 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/pt/

[UPD] Update document_page_approval.pot

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (62 of 62 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/it/

berisoft-arg committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (62 of 62 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/es/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (62 of 62 strings)

Translation: knowledge-16.0/knowledge-16.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_page_approval/sv/

EdgarRetes committed:
[MIG] document_page_approval: Migration to 17.0

victoralmau committed:
[IMP] document_page_approval: pre-commit auto fixes

victoralmau committed:
[MIG] document_page_approval: Finish migration

Changes done:
- Squash administrative commits
- Purge translations
- Remove ir.model.access.csv file
- Improve tests (change to setUpClass)
- Replace name_get() to _compute_display_name() (document_page)
- Remove readonly=1 (redudant, it already is) defined in approved_uid and approved_date fields in views
- Restore readonly=1 to the state field
- Remove duplicate Back to draft button

TT51529

Commit:
[UPD] Update document_page_approval.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-17.0/knowledge-17.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_page_approval/

Commit:
[UPD] Update document_page_approval.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-17.0/knowledge-17.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_page_approval/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (62 of 62 strings)

Translation: knowledge-17.0/knowledge-17.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_page_approval/it/

victoralmau committed:
Translated using Weblate (Spanish)

Currently translated at 98.3% (61 of 62 strings)

Translation: knowledge-17.0/knowledge-17.0-document_page_approval
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_page_approval/es/

BhaveshHeliconia committed:
[IMP] document_page_approval: pre-commit auto fixes

dreispt commented:
Can you please rebase on latest 18.0 - that should fix the tests.

BhaveshHeliconia commented:
@dreispt, Please Review. It's done.

dreispt commented on document_page_approval/models/document_page_history.py:
```diff
@@ -151,8 +151,9 @@ def _compute_page_url(self):
-                "{}/web#db={}&id={}&" "model=document.page.history"
-            ).format(base_url, self.env.cr.dbname, page.id)
+                f"{base_url}/web#db={self.env.cr.dbname}&id={page.id}&"
+                "model=document.page.history"
```
Nota that these URLs are still supported, but have changed and might stop working in future versions.

dreispt commented:
/ocabot migration document_page_approval

victoralmau committed:
[FIX] document_page_approval: Add Change Request Manager rule

Steps to reproduce the error:
- Modify the content of a page (with a different user).
- Go to the Change request menu and Cancel the change.
- Click on the "Back to draft" button.
- An error will be displayed

TT55569

victoralmau commented on document_page_approval/models/document_page_history.py:
```diff
@@ -156,14 +160,18 @@ def _compute_page_url(self):
-                rec.diff = self._get_diff(False, rec.id)
+                raw_diff = self._get_diff(False, rec.id)
+
+            cleaned_diff = re.sub(r'data-last-history-steps=".*?"', "", raw_diff)
+
+            rec.diff = cleaned_diff
```
What is the reason for this? Sorry, but I don't understand (if it is necessary, perhaps it would be appropriate to explain in the method the reasons, so that we all know it in the future).

victoralmau commented on document_page_approval/tests/test_document_page_approval.py:
```diff
@@ -9,18 +11,39 @@ def setUpClass(cls):
+        # Disable restrictive record rules for the test
+        cls.env["ir.rule"].search(
+            [("model_id", "=", cls.env["ir.model"]._get_id("document.page.history"))]
+        ).write({"active": False})
```
I don't understand why these changes are necessary. In migrations, the proper thing to do is to reduce the diff (for better readability).

BhaveshHeliconia replied:
we have to add this because of apply write access to document.page.history

victoralmau replied:
No, doing this in a test is not necessary for at least **99%**, and in this case it is not. Doing this indicates that something defined in the test is incorrect. Checking in detail I have already detected what happens, I suggest it now (remember, a test should test a real case, and there was some unnoticed but important step that was not being done).

victoralmau commented on document_page_approval/tests/test_document_page_approval.py:
```diff
@@ -0,0 +1,275 @@
+        cls.page1 = cls.env.ref("document_page.demo_page1")
+
+        # Create test user without groups first
+        cls.user2 = new_test_user(cls.env, login="test-user2", groups="base.group_user")
+
+        # Ensure user2 has the approver group
+        cls.approver_gid = cls.env.ref(
+            "document_page_approval.group_document_approver_user"
+        )
+        cls.user2.write({"groups_id": [Command.link(cls.approver_gid.id)]})
```
```suggestion
        cls.user2 = new_test_user(cls.env, login="test-user2", groups="document_page_approval.group_document_approver_user")

        cls.approver_gid = cls.env.ref(
            "document_page_approval.group_document_approver_user"
        )
```
The `base.group_user` permission is inherited indirectly with `document_knowledge`

BhaveshHeliconia commented:
@victoralmau, Please Review.it's done.

victoralmau commented on document_page_approval/tests/test_document_page_approval.py:
```diff
@@ -103,41 +127,153 @@ def test_change_request_from_scratch(self):
-        self.assertNotEqual(page.approved_date, chreq.approved_date)
-        self.assertNotEqual(page.approved_uid, chreq.approved_uid)
 
-        chreq.action_to_approve()
```
It is not necessary to remove it, it is important.

victoralmau commented on document_page_approval/tests/test_document_page_approval.py:
```diff
@@ -103,41 +127,153 @@ def test_change_request_from_scratch(self):
 
-        chreq.action_to_approve()
+        # Move change request to 'to approve' state
+        chreq.with_user(self.user2).action_to_approve()
```
This is not necessary, it is already in “to approve” status requested by the user who created the history.

victoralmau commented on document_page_approval/tests/test_document_page_approval.py:
```diff
@@ -103,41 +127,153 @@ def test_change_request_from_scratch(self):
 
-        chreq.action_approve()
+        # Approve the change request
+        chreq.with_user(self.user2).action_approve()
```
If you want `user2` to approve it, you need it to be in “to approve” status, so you will need to do the following first

```
chreq.action_to_approve()
self.assertEqual(chreq.state, "to approve")
```

victoralmau reviewed (changes requested):
Now you can remove the code from https://github.com/OCA/knowledge/pull/524#discussion_r2012509984

victoralmau commented:
Please, cherry-pick https://github.com/OCA/knowledge/pull/531 to commit history before migration commit.

BhaveshHeliconia commented:
@victoralmau, Please Review.i updated test cases.

BhaveshHeliconia committed:
[MIG] document_page_approval: Migration to 18.0

victoralmau reviewed (approved):
Thank you, good job.

Remember for the future that it is not necessary to add line breaks. I comment this because it helps to reduce the diff of the migration commit and is more readable.

pedrobaeza commented:
/ocabot merge nobump
